distributed-commandbus-springcloud/src/main/java/org/axonframework/springcloud/commandhandling/SpringCloudHttpBackupCommandRouter.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2010-2017. Axon Framework
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.springcloud.commandhandling;
+
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.commandhandling.distributed.RoutingStrategy;
+import org.axonframework.commandhandling.distributed.SimpleMember;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.function.Predicate;
+
+/**
+ * Implementation of the {@link org.axonframework.springcloud.commandhandling.SpringCloudCommandRouter} which has a
+ * backup mechanism to request and provide a member and/or its {@link MessageRoutingInformation}.
+ * It uses {@link org.springframework.web.client.RestTemplate} to request {@code MembershipInformation} and services the
+ * purposes of a queryable location to retrieve the local {@code MembershipInformation} from by being a
+ * {@link org.springframework.web.bind.annotation.RestController}.
+ *
+ * @author Steven van Beelen
+ */
+@RestController
+@RequestMapping(SpringCloudHttpBackupCommandRouter.MESSAGE_ROUTING_INFORMATION_PATH)
```

Comment:
It would be nice if the URL refers to a property that can be configured, with a default in case it isn't.
Check out how Spring Boot registers the H2 console based on the availability of certains property settings:
`org.springframework.boot.autoconfigure.h2.H2ConsoleAutoConfiguration`.

In our case, it would be nice to have some properties like:
- axon.distributed.spring-cloud-discovery.endpoint-info.fallback-to-http-get (default to true)
- axon.distributed.spring-cloud-discovery.endpoint-info.fallback-url (default to /message-routing-information)

If the first property is set to false, the autoconfiguration will return a "SpringCloudCommandRouter" instance, otherwise a "ServletRegistrationBean".

I am not sure you can use a property as the URL in a `@RequestMapping`. If you can't you can use the mechanism as defined in the aforementioned `H2ConsoleAutoConfiguration` class (using a ServletRegistrationBean).

Note that the `ServletRegistrationBean` should only be configured for web-applications (use `@ConditionalOnWebApplication`)

Reply:
This is a great idea, gonna implement it

Reply:
Adjusted the set up such that a property is used to set the request mapping and the endpoint used to perform the HTTP GET request with.
This property is set in the newly introduced `SpringCloudProperties` in the `DistributedCommandBusProperties` class.
If a user isn't using the `axon-spring-boot-started` dependency, he'll have to set the property in it's configuration file itself, or otherwise it's defaulted to `/message-routing-information`.